### PR TITLE
Add v3 node groups

### PR DESCRIPTION
**Pull request by illegalnumbers**

### CODE REVIEW

**Comment by illegalnumbers** [comment]
refs https://github.com/streamnative/argo/issues/371

**Comment by illegalnumbers** [comment]
See here https://github.com/streamnative/argo/blob/20a10b6c4a26ec4142383b39305a520a3510e8af/workflowtemplates/aws-pm/provision2.tpl.tf#L112-L124

File: main.tf [hunk 1]
```diff
@@ -57,8 +56,12 @@ locals {
     "8xlarge" = "Large"
   }
 
+  v3_compute_units = {
+    "large" = "Small"
```
Comment (EronWright) [inline main.tf:60]: Seems weird, any idea what this means? `"large" = "Small"`

**Review by EronWright — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (illegalnumbers) [reply main.tf:60]: Yes, I actually had this same question for Addison. There was an unfortunate prior naming convention for compute units it seems that he and Joey had came up with. See [here](https://streamnative.slack.com/archives/C04AA1AV274/p1686688929002099)

File: main.tf [hunk 2]
```diff
@@ -103,18 +106,48 @@ locals {
...
+  } : {}
+
+  v3_node_groups = tomap({
+    "core" = {
+      subnet_ids     = var.private_subnet_ids
+      instance_types = [var.v3_node_group_core_instance_type]
+      name           = "snc-core"
+      taints         = local.v3_node_taints
+      desired_size   = 3
+      min_size       = 3
```
Comment (maxsxu) [inline main.tf:141]: Would it better to set the desired_size and min_size to var.node_pool_desired_size and var.node_pool_min_size, or the length of private_subnet_ids?

File: variables.tf [hunk 3]
```diff
@@ -227,6 +227,12 @@ variable "enable_istio" {
   type        = bool
 }
 
+variable "enable_resource_creation" {
```
Comment (maxsxu) [inline variables.tf:230]: Would it better to reuse the existing variable `enable_bootstrap` rather than create a new one?

File: variables.tf [hunk 4]
```diff
@@ -329,6 +353,12 @@ variable "kiali_operator_settings" {
   type        = map(any)
 }
 
+variable "manage_aws_auth_configmap" {
```
Comment (maxsxu) [inline variables.tf:356]: Wondering in which case we would set this variable to false?

File: main.tf [hunk 5]
```diff
@@ -103,18 +106,48 @@ locals {
...
+  v3_node_taints = var.enable_v3_node_taints ? {
+    "core" = {
+      key    = "node.cloud.streamnative.io/core"
+      value  = "true"
+      effect = "NO_SCHEDULE"
+    }
+  } : {}
+
+  v3_node_groups = tomap({
+    "core" = {
```
Comment (maxsxu) [inline main.tf:135]: Would be better to keep consistent with the `name`

```suggestion
    "snc-core" = {
```

File: main.tf (hunk 2 above)
Comment (illegalnumbers) [reply main.tf:141]: seems fine to me, i dont have context as to why addison hard coded those

File: variables.tf (hunk 3 above)
Comment (illegalnumbers) [reply variables.tf:230]: that was intentional, there's some messaging in the commit on this line regarding it but their distinct for a reason (using at different provisioning steps in the workflows)

File: variables.tf (hunk 4 above)
Comment (illegalnumbers) [reply variables.tf:356]: there's little context on this anywhere, so i'm not going to remove it but here's some slack messaging in the past regarding this https://streamnative.slack.com/archives/CVDQE4338/p1664320203538319?thread_ts=1664305250.471299&cid=CVDQE4338

File: main.tf (hunk 5 above)
Comment (illegalnumbers) [reply main.tf:135]: i think updating this for consumers of the module after this commit will likely add in another node group and/or remove the one they have so we'll have to be careful applying this in the future to things already using it. but i'm fine with the change.

**Review by ignacioli — APPROVED** [review]

**Review by maxsxu — APPROVED** [review]

**Pull request merged**